packages/vats/src/core/types.ts

```diff
@@ -0,0 +1,511 @@
+  provisioning: Awaited<ProvisioningVat> | undefined;
+  provisionBridgeManager: import('../types.js').ScopedBridgeManager | undefined;
+  provisionWalletBridgeManager:
+    | import('../types.js').ScopedBridgeManager
+    | undefined;
+  storageBridgeManager: import('../types.js').ScopedBridgeManager | undefined;
+  /**
+   * Convienence function for starting a contract (ungoverned) and saving its
+   * facets (including adminFacet)
+   */
+  startUpgradable: StartUpgradable;
+  /** kits stored by startUpgradable */
+  contractKits: MapStore<Instance, StartedInstanceKitWithLabel>;
+  /** Convience function for starting contracts governed by the Econ Committee */
+  startGovernedUpgradable: StartGovernedUpgradable;
+  /** kits stored by startGovernedUpgradable */
+  governedContractKits: MapStore<
+    Instance,
+    GovernanceFacetKit<any> & { label: string }
+  >;
+  /** Used only for testing. Should not appear in any production proposals. */
+  testFirstAnchorKit: import('../vat-bank.js').AssetIssuerKit<'nat'>;
+  walletBridgeManager: import('../types.js').ScopedBridgeManager | undefined;
+  walletFactoryStartResult: import('./startWalletFactory').WalletFactoryStartResult;
+  provisionPoolStartResult: unknown;
+  vatStore: import('./utils.js').VatStore;
+  zoe: ZoeService;
+};
+
+type ChainBootstrapSpace = PromiseSpaceOf<ChainBootstrapSpaceT>;
+
+type BootstrapVatParams = {
+  argv: {
+    hardcodedClientAddresses?: string[];
+    FIXME_GCI: string;
+    PROVISIONER_INDEX?: number;
+  };
+};
+
+type BootstrapPowers = BootstrapSpace & {
+  zone: import('@agoric/zone').Zone;
+  devices: SoloDevices | ChainDevices;
+  vats: SwingsetVats;
+  vatPowers: { [prop: string]: any; D: DProxy };
+  vatParameters: BootstrapVatParams;
+  runBehaviors: (manifest: unknown) => Promise<unknown>;
+  modules: Record<string, Record<string, any>>;
+};
+
+type BootstrapSpace = WellKnownSpaces &
+  PromiseSpaceOf<
+    ChainBootstrapSpaceT & {
+      vatAdminSvc: VatAdminSvc;
+    },
+    {},
+    {
+      loadVat: VatLoader<unknown>;
+      loadCriticalVat: VatLoader<unknown>;
+    }
+  >;
+
+type ProvisioningVat = ERef<
+  ReturnType<typeof import('../vat-provisioning.js').buildRootObject>
+>;
+
+type MintsVat = ERef<
+  ReturnType<typeof import('../vat-mints.js').buildRootObject>
+>;
+
+type PriceAuthorityVat = ERef<
+  ReturnType<typeof import('../vat-priceAuthority.js').buildRootObject>
+>;
+
+type NetworkVat = ERef<
+  ReturnType<typeof import('../vat-network.js').buildRootObject>
+>;
+type IBCVat = ERef<ReturnType<typeof import('../vat-ibc.js').buildRootObject>>;
+type NamedVatPowers = {
+  namedVat: PromiseSpaceOf<{
+    agoricNames: Awaited<AgoricNamesVat>;
+    board: Awaited<BoardVat>;
+  }>;
+};
+
+type RemoteIssuerKit = {
+  mint: ERef<Mint>;
+  issuer: ERef<Issuer>;
+  brand: Brand;
+};
+type AgoricNamesVat = ERef<
+  ReturnType<typeof import('../vat-agoricNames').buildRootObject>
+>;
```

Comment:
Does typescript allow one to make shorthand for the oft-repeated `ERef<ReturnType<typeof importedThing.buildRootObject>>`?

Reply:
Yes, good idea! We can have a utility type for the root object of a vat.

Reply:
I tried DRYing out the `buildRootObject` reference but I haven't found a way to get the module import to be a generic type for s helper. It might require a module import, which would break this being an ambient.

We could DRY the `ERef<ReturnType<` but it wouldn't save a lot of characters unless we also do a helper like `ContractInstallationPromises` (which I just added). I expect eventually we'll want one for devices to be keys in a namespace instead of named by concatenation